api/hypershift/v1beta1/aws.go

```diff
@@ -25,6 +25,17 @@ type AWSNodePoolPlatform struct {
 	// +kubebuilder:validation:MaxLength=255
 	AMI string `json:"ami,omitempty"`
 
+	// imageType specifies the type of image to use for node instances.
+	// Valid values are ImageTypeLinux ("Linux") or ImageTypeWindows ("Windows").
+	// When set to ImageTypeWindows, the controller will automatically populate the AMI field
+	// with a Windows-compatible AMI based on the region and OpenShift version.
+	// When the AMI field is also set, it will take precedence over automatic selection based
+	// on imageType. Also note that Windows ImageType is only compatible with an Arch of amd64
```

Comment:
This means that once the AMI is set, it is never updated right? Does hypershift have any concept of boot image upgrades automatically being applied? Would the boot image update controller know that this must be a windows image?